libs/types/src/solana/mod.rs

```diff
@@ -32,30 +33,57 @@ pub struct SendTransactionParams {
 }
 
 impl SendTransactionParams {
+    /// Parameters for a `sendTransaction` request with the given transaction already encoded wit
+    /// the given encoding.
+    pub fn from_encoded_transaction(
+        transaction: String,
+        encoding: SendTransactionEncoding,
+    ) -> Self {
+        Self {
+            transaction,
+            encoding: Some(encoding),
+            skip_preflight: None,
+            preflight_commitment: None,
+            max_retries: None,
+            min_context_slot: None,
+        }
+    }
+
     /// Returns `true` if all of the optional config parameters are `None` and `false` otherwise.
     pub fn is_default_config(&self) -> bool {
-        self.encoding.is_none()
-            && self.skip_preflight.is_none()
-            && self.preflight_commitment.is_none()
-            && self.max_retries.is_none()
-            && self.min_context_slot.is_none()
+        let SendTransactionParams {
+            transaction: _,
+            encoding,
+            skip_preflight,
+            preflight_commitment,
+            max_retries,
+            min_context_slot,
+        } = &self;
+        encoding.is_none()
+            && skip_preflight.is_none()
+            && preflight_commitment.is_none()
+            && max_retries.is_none()
+            && min_context_slot.is_none()
+    }
+
+    /// The transaction being sent as an encoded string.
+    pub fn get_transaction(&self) -> String {
+        self.transaction.clone()
+    }
```

Comment:
nit: using something by reference and cloning is an anti-pattern. Let the caller decide whether cloning is necessary
```suggestion
    pub fn get_transaction(&self) -> &str {
        &self.transaction
    }
```